Landmine Rework

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Comprehensive rework of landmines.

- [x] Turn mines into an item
- [x] Port a bunch of tg improvements
- [x] Port tg shrapnel fixes
- [x] Add inhand mine deploying
- [x] Mines now take time to arm
- [x] Add mine disarming
- [x] Delayed explosion
- [x] fix shrapnel runtime 
- [x] Actually MOVE the Mines file to code/items
- [x] Write a bunch of new mine types
- [x] Implement new mines in spawners
- [x] YOU THOUGHT IT WAS OVER. Proximity Mines & full pressure mine code edit
- [x] Embeds now work as intended

- [ ] Sprites (need help)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Pipe bomb....
Crew mortality grew two sizes that day
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
add: Redesigns landmines as an item that can be picked up, disarmed, deployed. Some logic courtesy of /tg/.
add: Adds a proximity explosive variant of landmines.
fix: shrapnel embedding once again respects targeted limb and no longer runtimes on hitting a wall or destroying an object.
fix: storage can no longer pick up anchored objects, causing them to get stuck inside
balance: Nerfs the ability of heavy explosions to immediately breach to space
soundadd: adds two mine detonation noises
code: moves weapon manufacturer defines to a DEFINE file
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
woa... pipe bomb...

mines pass 1

Sun-Soaked committed:
more stuff

tg port fixes shrapnel crashes(hopefully)

the mines on wasteplanet are free, I have 17 at home

mine behavior

Sun-Soaked committed:
fixes & hair trigger mines

runtime gonb
adds functionality for mines that explode immediately(like old times)
some linter stuff
early mine type organization

Sun-Soaked committed:
handling

you can no longer trigger a mine immediately after placing it

Sun-Soaked committed:
alert anim & linters

Sun-Soaked committed:
oops

Sun-Soaked committed:
Merge branch 'master' into landmines

west3436 commented on code/game/objects/effects/mines.dm:
```diff
@@ -1,162 +1,433 @@
+
+//handles disarming(and failing to disarm)
+/obj/item/mine/attackby(obj/item/I, mob/user)
+	if(armed)
```
Save an indent level via:
`if(!armed)`
`     to_chat(user, span_notice("You hit \the [src] with [I]. Thankfully, nothing happens."))`
`return`

Sun-Soaked replied:
true

Sun-Soaked committed:
yeah

Sun-Soaked committed:
disarming

real disarming & mine hand interactions make sense now

Sun-Soaked committed:
fixes pathing

we LOVE pathing

Sun-Soaked committed:
fixes

Sun-Soaked committed:
gyuh???

assorted dumb stuff

Sun-Soaked committed:
manufacturers file & mine stuff

g

Sun-Soaked committed:
why...

why does this happen

Sun-Soaked committed:
fixes hair trigger, mine exit slowdown

video james?

also some wip mine subtypes

Sun-Soaked committed:
mining away...

incin mine rework

plasma and radmines

some reorg

sealed var for undisarmable mines

Sun-Soaked committed:
reorg, triggershred fix, shrap buffs

gaming

Sun-Soaked committed:
nerfs heavy explosion breaching power

I will be asked to atomize this. I know it

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
bugfixathon

opacity fixed
click fixed
disarming fixed
det time tweak

Sun-Soaked committed:
forgot this

Sun-Soaked committed:
fixensmoken

-adds new wire defines that have more accurate names
-fixes mines failing to recover from dud when their trigger wire is mended
-fixes dud mines not flagging being stepped on
-refactors mine slowdown(into being functional, as opposed to not)

Sun-Soaked committed:
nerf + refactors + linter fix + spawners

adds live mine helper

mine spawners

spawner mine
chain explosions shouldn't be likely, allows for minefields

refactors disarming to a proc

removes junk var I'm not using anymore

shrap nerf now that shred-triggerer is back

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
mine slowdown nerf + fireshrap buff + ruin mines

Sun-Soaked committed:
mine implementation + training mine

god have mercy on your souls

Sun-Soaked committed:
sparkle on! don't forget to be

checks get angry if an addtimer's timer is 0 ticks

Sun-Soaked committed:
reorganizes mine file + desc change

also hopefully this catches the bug from earlier

hair trigger flagging in desc

Sun-Soaked committed:
fix 2

should satisfy create_and_destroy()

Sun-Soaked committed:
maybe?

Sun-Soaked committed:
Merge remote-tracking branch 'upstream/master' into landmines

Sun-Soaked committed:
Merge branch 'master' into landmines

Signed-off-by: Sun-Soaked <45698967+Sun-Soaked@users.noreply.github.com>

Sun-Soaked committed:
OOPS!

Sun-Soaked committed:
huh?

Sun-Soaked committed:
okay what

Sun-Soaked committed:
missed this

also I need the git to redo checks

Sun-Soaked commented:
should be feature complete now. Just needs sprites and I'll undraft

Sun-Soaked committed:
yippee!!!

thgvr commented on code/game/objects/items/devices/mines.dm:
```diff
@@ -0,0 +1,657 @@
+	visible_message("<span class='danger'>\The [src] beeps softly, indicating it is now active.<span>", vision_distance = COMBAT_MESSAGE_RANGE)
+
+//insert your horrible fate here
+/obj/item/mine/proc/mineEffect(mob/victim)
```
```suggestion
/obj/item/mine/proc/mine_effect(mob/victim)
```

thgvr commented on code/game/objects/items/devices/mines.dm:
```diff
@@ -0,0 +1,657 @@
+	foot_on_mine = null
+
+/// When something sets off a mine
+/obj/item/mine/proc/triggermine(atom/movable/triggerer)
```
```suggestion
/obj/item/mine/proc/trigger_mine(atom/movable/triggerer)
```

thgvr commented on code/game/objects/items/devices/mines.dm:
```diff
@@ -0,0 +1,657 @@
+//using an unarmed mine inhand deploys it.
+/obj/item/mine/attack_self(mob/user)
+	if(!armed)
+		user.visible_message("<span class='danger'>[user] deploys the [src].</span>", "<span class='notice'>You deploy the [src].</span>")
```
```suggestion
		user.visible_message(span_danger("[user] deploys the [src].</span>"), span_notice("You deploy the [src].</span>"))
```

Sun-Soaked committed:
sooder reviews + removes fuse wire + stingball buff

e

Sun-Soaked committed:
crushes u in my fist

maps...

Sun-Soaked committed:
PROXIMITY MINE MEGAREWORK

WHO ELSE UP FULLY REFACTORING THEY PR 3 HOURS BEFORE UNDRAFTING IT

Sun-Soaked committed:
stinger proximity

plus assorted tweaks

Sun-Soaked committed:
whoopsies

Sun-Soaked committed:
oopsie daisy

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
stuff

wording fix + warning light for armed proximity mines(I'll make this nicer later, pinkie promise)

Sun-Soaked committed:
Merge branch 'landmines' of https://github.com/Sun-Soaked/Shiptest into landmines

Sun-Soaked committed:
professional grade stupidity

Sun-Soaked committed:
slow explosion wire + fixes

the lesser failure to match the lesser success

Sun-Soaked committed:
shrap rebalance + mine stall increase

shrap should disembowl people less and dismember them more

mine freeze is longer

Sun-Soaked committed:
some fixes and stuff and fixes and

shrapnel now works properly

Sun-Soaked committed:
throwing obj block + training mines

training mine is (less) shitcode & now disarms the existing object

adds functionality for reusable mines(set trigger to false in mine_effect)

items thrown past a mine no longer instantly detonate it

Sun-Soaked committed:
makes master compat

Sun-Soaked committed:
Merge remote-tracking branch 'upstream/master' into landmines

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
anchor fix + handling picked up mines

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
removes unused movespeed stuff

whoopsies

Sun-Soaked committed:
yeah.

stealth "mechanic" removal

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked committed:
tweaks

admin message
fixes prox mines(?)
moves landmines to their own file

Sun-Soaked committed:
Merge remote-tracking branch 'upstream/master' into landmines

Sun-Soaked committed:
Merge branch 'master' into landmines

Sun-Soaked commented:
todo in future pr: more fitting sprites

Sun-Soaked committed:
Merge branch 'master' into landmines
